### PR TITLE
p2p(ticdc): use a larger max receive message size

### LINKED ISSUE #8004: p2p: grpc: received message larger than max
Checkpoint can not advance when replicating 72k tables in v6.5.0.

```
[2022/12/30 17:01:00.000 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=379a741e-154d-4863-a15c-da459813c5d1] [senderAdvertiseAddr=172.16.5.37:48112] [addr=172.16.5.37:57070] [epoch=372]
[2022/12/30 17:01:00.006 +08:00] [WARN] [server.go:630] ["peer-to-peer message handler error"] [error="rpc error: code = ResourceExhausted desc = grpc: received message larger than max (4207687 vs. 4194304)"] [errorVerbose="rpc error: code = ResourceExhausted desc = grpc: received message larger than max (4207687 vs. 4194304)\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).receive\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:722\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).SendMessage.func1\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:629\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
[2022/12/30 17:01:00.006 +08:00] [WARN] [server.go:633] ["error receiving from peer"] [error="context canceled"]
[2022/12/30 17:01:00.025 +08:00] [WARN] [server.go:334] ["sending response to peer failed"] [error="[CDC:ErrPeerMessageInternalSenderClosed]peer-to-peer message server tries to send to a closed stream. Internal only."] [errorVerbose="[CDC:ErrPeerMessageInternalSenderClosed]peer-to-peer message server tries to send to a closed stream. Internal only.\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/p2p.(*streamHandle).Send\n\tgithub.com/pingcap/tiflow/pkg/p2p/server_stream_handle.go:66\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).tick\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:329\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).run\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:228\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).Run.func2\n\tgithub.com/pingcap/tiflow/pkg/p2p/server.go:206\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
[2022/12/30 17:01:00.025 +08:00] [INFO] [server.go:343] ["Deregistering peer"] [sender=379a741e-154d-4863-a15c-da459813c5d1] [epoch=372] []
[2022/12/30 17:01:00.391 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=0dc8a568-2e7c-4f35-b6fa-a5d3991cf1c6] [senderAdvertiseAddr=172.16.5.37:48114] [addr=172.16.5.37:57074] [epoch=406]
```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #8004

### What is changed and how it works?
 
Use a larger max receive message size to address ResourceExhausted issue.

#### Tests 

 - Unit test
 - Manual test
 	Setup a changefeed that replicate 72k tables.

image

##### Will it cause performance regression or break compatibility?

No.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
Fix an issue that checkpoint can not advance when replicating many tables.
```

### CODE REVIEW

File: pkg/config/messages.go [hunk 1]
```diff
@@ -121,6 +127,14 @@ func (c *MessagesConfig) ValidateAndAdjust() error {
 		return cerrors.ErrInvalidServerOption.GenWithStackByArgs("server-worker-pool-size is larger than 32")
 	}
 
+	if c.MaxRecvMsgSize == 0 {
+		c.MaxRecvMsgSize = defaultMaxRecvMsgSize
+	}
+	if c.MaxRecvMsgSize <= 0 {
```
Comment (amyangfei) [inline pkg/config/messages.go:133]:
```suggestion
	if c.MaxRecvMsgSize < 0 {
```

**Comment by lance6716** [comment]
maybe this can help

https://github.com/pingcap/tiflow/blob/73bb333c8b8319ffcb402b676fefaf7bd51e1942/dm/ctl/common/util.go#L124-L133

File: pkg/config/messages.go (hunk 1 above)
Comment (3AceShowHand) [reply pkg/config/messages.go:133]: It looks this 2 case can be switched, check `<0` and return the error first, then check if it's `0`.

**Review by amyangfei — APPROVED** [review]

File: pkg/cmd/server/server_test.go [hunk 2]
```diff
@@ -205,6 +205,7 @@ func TestParseCfg(t *testing.T) {
 				ServerMaxPendingMessageCount: 102400,
 				ServerAckInterval:            config.TomlDuration(time.Millisecond * 100),
 				ServerWorkerPoolSize:         4,
+				MaxRecvMsgSize:               268435456,
```
Comment (3AceShowHand) [inline pkg/cmd/server/server_test.go:208]:
```suggestion
				MaxRecvMsgSize:               256 * 1024 * 1024,
```

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**